InsanityTests/GeneratorSpec.swift

```diff
@@ -46,11 +47,27 @@ class GeneratorSpec: QuickSpec {
             }
 
             it("feeds types.implementing specific protocol") {
-                expect(generate("Found {{ types.implementing.Decodable.count }} types")).to(equal("Found 1 types"))
+                expect(generate("Found {{ types.implementing.KnownProtocol.count }} types")).to(equal("Found 2 types"))
+                expect(generate("Found {% ifnot types.implementing.Decodable %}0{% endif %} types")).to(equal("Found 0 types"))
+                expect(generate("Found {% ifnot types.implementing.Foo %}0{% endif %} types")).to(equal("Found 0 types"))
+                expect(generate("Found {% ifnot types.implementing.NSObject %}0{% endif %} types")).to(equal("Found 0 types"))
+                expect(generate("Found {% ifnot types.implementing.Bar %}0{% endif %} types")).to(equal("Found 0 types"))
             }
 
-            it("feeds types.inheriting specific protocol") {
-                expect(generate("Found {{ types.inheriting.Decodable.count }} types")).to(equal("Found 1 types"))
+            it("feeds types.inheriting specific class") {
+                expect(generate("Found {{ types.inheriting.Foo.count }} types")).to(equal("Found 1 types"))
+                expect(generate("Found {% ifnot types.inheriting.NSObject.count %}0{% endif %} types")).to(equal("Found 0 types"))
+                expect(generate("Found {% ifnot types.inheriting.Bar %}0{% endif %} types")).to(equal("Found 0 types"))
+                expect(generate("Found {% ifnot types.inheriting.KnownProtocol %}0{% endif %} types")).to(equal("Found 0 types"))
+                expect(generate("Found {% ifnot types.inheriting.Decodable %}0{% endif %} types")).to(equal("Found 0 types"))
+            }
+
+            it("feeds types.based specific type or protocol") {
+                expect(generate("Found {{ types.based.Foo.count }} types")).to(equal("Found 1 types"))
```

Comment:
could you re-order those 3 containers to have the same the order of types?, that way it's easier to read through